logicrunner/handle_calls.go

```diff
@@ -158,6 +157,30 @@ func (h *HandleCall) handleActual(
 	if objRef == nil {
 		return nil, errors.New("can't get object reference")
 	}
+	if !objRef.Record().Equal(reqInfo.ObjectID) {
+		return nil, errors.New("object id we calculated doesn't match ledger")
+	}
+
+	repl := &reply.RegisterRequest{Request: *requestRef}
+
+	if reqInfo.Request != nil && len(reqInfo.Request) != 0 {
```

Comment:
2 ifs with duplicate checks

Reply:
these are not exact duplicates, but left only one check